test/test.js

```diff
@@ -13,11 +13,32 @@ describe('session-resume', function () {
 
   describe('restoreResumableFields', function () {
     it('restores fields values from session storage', function () {
-      sessionStorage.setItem('session-resume:test-persist', JSON.stringify([['my-first-field', 'test2']]))
+      sessionStorage.setItem(
+        'session-resume:test-persist',
+        JSON.stringify([
+          ['my-first-field', 'test2'],
+          ['non-existant-field', 'test3']
+        ])
+      )
       restoreResumableFields('test-persist')
 
       assert.equal(document.querySelector('#my-first-field').value, 'test2')
       assert.equal(document.querySelector('#my-second-field').value, 'second-field-value')
+
+      // Some fields we want to restore are not always present in the DOM
+      // and may be added later. We hold onto the values until they're needed.
+      assert.deepEqual(JSON.parse(sessionStorage.getItem('session-resume:test-persist')), [
+        ['non-existant-field', 'test3']
+      ])
+    })
```

Comment:
Instead of changing the existing test, I'd feel more comfortable if we left the current tests unmodified where possible and create new tests for the functionality we are adding.

Reply:
Roger that. I went ahead and adjusted the tests so that the original tests remain unchanged: https://github.com/github/session-resume/pull/16/commits/31a22b51bb1a18068d432366fb307f1d5d1f3107